src/node.ts

```diff
@@ -56,7 +59,38 @@ export abstract class BaseNode<
    */
   readonly children: FunctionlessNode[] = [];
 
-  constructor(readonly kind: Kind) {}
+  constructor(readonly kind: Kind, args: IArguments) {
+    const setParent = (node: any) => {
+      if (!node) {
+        return;
+      } else if (isNode(node)) {
+        node.setParent(this as FunctionlessNode);
+      } else if (Array.isArray(node)) {
+        node.forEach(setParent);
+      }
+    };
+    for (const arg of args) {
+      setParent(arg);
+    }
```

Comment:
I don't understand, what's wrong?

Reply:
Nothing wrong, just saying your function also supports arrays of arguments so it could be written as:

```ts
const setParent = (node: any) => {
      if (!node) {
        return;
      } else if (isNode(node)) {
        node.setParent(this as FunctionlessNode);
      } else if (Array.isArray(node)) {
        node.forEach(setParent);
      }
    };
setParent(args);
```

Reply:
Actually, IArguments is not an array, it is only Iterable

Reply:
Let me take another look, perhaps I got confused by a type error. I do remember doing it your way to begin with